test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -1138,4 +1148,30 @@ default void run()
             }
         }
     }
+    private void logSystemInfo()
+    {
+        if (inInstancelogger.isInfoEnabled())
+        {
+            try
+            {
+                inInstancelogger.info("Hostname: {}", InetAddress.getLocalHost().getHostName() + ":" + DatabaseDescriptor.getStoragePort() + ":" + DatabaseDescriptor.getSSLStoragePort());
+            }
+            catch (UnknownHostException e1)
+            {
+                inInstancelogger.info("Could not resolve local host");
+            }
+
+            inInstancelogger.info("JVM vendor/version: {}/{}", JAVA_VM_NAME.getString(), JAVA_VERSION.getString());
+            inInstancelogger.info("Heap size: {}/{}",
+                        FBUtilities.prettyPrintMemory(Runtime.getRuntime().totalMemory()),
+                        FBUtilities.prettyPrintMemory(Runtime.getRuntime().maxMemory()));
+
+            for(MemoryPoolMXBean pool: ManagementFactory.getMemoryPoolMXBeans())
+                inInstancelogger.info("{} {}: {}", pool.getName(), pool.getType(), pool.getPeakUsage());
+
+            inInstancelogger.info("Classpath: {}", JAVA_CLASS_PATH.getString());
+
+            inInstancelogger.info("JVM Arguments: {}", ManagementFactory.getRuntimeMXBean().getInputArguments());
+        }
```

Comment:
rather than mimic we should refactor so we have a function that CassandraDaemon and Instance both call.

Reply:
Yeah, just make `CassandraDaemon#logSystemInfo()` static and parameterize on the `Logger`.